docs/2-building-blocks-of-oop-part-1/hometask/src/Triangle.ts

```diff
@@ -1,3 +1,56 @@
 import { Shape } from './Shape';
+import { Point } from './Point';
 
-export class Triangle extends Shape {}
+export class Triangle extends Shape {
+  constructor(point1: Point, point2: Point, point3: Point);
+  constructor(
+    point1: Point,
+    point2: Point,
+    point3: Point,
+    color: string,
+    filled: boolean
+  );
+  constructor(
+    point1: Point,
+    point2: Point,
+    point3: Point,
+    color?: string,
+    filled?: boolean
+  ) {
+    if (typeof color === 'string' && typeof filled === 'boolean') {
+      super([point1, point2, point3], color, filled);
+    } else {
+      super([point1, point2, point3]);
+    }
+  }
+
+  toString() {
+    return `Triangle[${this.points
+      .map((point, index) => `v${index + 1}=${point.toString()}`)
+      .join(',')}]`;
+  }
+
+  getType(): string {
+    const distances = this.points.map((point, index, points) => {
```

Comment:
It would be better to divide this logic in separated method and use it in `this.getPerimeter()` and here to avoid duplicated code )

Reply:
moved to a separate method.